DuckDuckGo/NetworkProtectionInviteViewModel.swift

```diff
@@ -19,67 +19,72 @@
 
 #if NETWORK_PROTECTION
 
-import NetworkProtection
 import Combine
-import Common
+import NetworkProtection
+
+enum NetworkProtectionInviteStep {
+    case codeEntry, success
+}
 
 final class NetworkProtectionInviteViewModel: ObservableObject {
+    @Published var currentStep: NetworkProtectionInviteStep = .codeEntry
     @Published var text: String = "" {
         didSet {
             if oldValue != text {
                 text = text.uppercased()
+                shouldDisableSubmit = text.count == 0
             }
         }
     }
-    @Published var redeemedText: String?
-    @Published var errorText: String?
+    var errorText: String = ""
+    @Published var shouldShowAlert: Bool = false
+    @Published var shouldDisableSubmit: Bool = true
 
-    private let tokenStore: NetworkProtectionTokenStore
     private let redemptionCoordinator: NetworkProtectionCodeRedeeming
-    private let featureVisibility: NetworkProtectionFeatureVisibility
-    private var textCancellable: AnyCancellable?
-
-    private static let errorEvents: EventMapping<NetworkProtectionError> = .init { _, _, _, _ in
-
-    }
+    private let completion: () -> Void
 
-    init() {
-        let tokenStore = NetworkProtectionKeychainTokenStore(useSystemKeychain: false,
-                                                             errorEvents: nil)
-        self.tokenStore = tokenStore
-        self.redemptionCoordinator = NetworkProtectionCodeRedemptionCoordinator(tokenStore: tokenStore,
-                                                                                errorEvents: Self.errorEvents)
-        self.featureVisibility = tokenStore
-        updateAuthenticatedText()
+    init(redemptionCoordinator: NetworkProtectionCodeRedeeming, completion: @escaping () -> Void) {
+        self.completion = completion
+        self.redemptionCoordinator = redemptionCoordinator
     }
 
     @MainActor
     func submit() async {
-        errorText = nil
         do {
             try await redemptionCoordinator.redeem(text.trimmingWhitespace())
-            updateAuthenticatedText()
         } catch NetworkProtectionClientError.invalidInviteCode {
-            errorText = "Unrecognized invite code"
+            errorText = UserText.inviteDialogUnrecognizedCodeMessage
+            shouldShowAlert = true
             return
         } catch {
-            errorText = "Error: try again"
+            errorText = UserText.unknownErrorTryAgainMessage
+            shouldShowAlert = true
+            return
         }
+        currentStep = .success
+    }
+
+    func getStarted() {
+        completion()
     }
 
+    // MARK: Dev only. Will be removed during https://app.asana.com/0/0/1205084446087078/f
```

Comment:
I was using this same ViewModel for the basic invite code entry on the status view. Rather than move it in there, I will keep these functions here for now until I’ve added a replacement, probably in the debug menu, as part of the status view subtask.